packages/clay-core/src/table/Body.tsx

```diff
@@ -20,14 +21,66 @@ type Props<T> = {
 	items?: Array<T>;
 } & React.TableHTMLAttributes<HTMLTableSectionElement>;
 
+type ItemProps<T> = {
+	children: React.ReactElement;
+	item: T;
+	keyValue: React.Key;
+};
+
+function* flatten<T extends Record<string, any>>(
+	array: Array<T>,
+	nestedKey: string,
+	level = 1
+): IterableIterator<T> {
+	for (let i = 0; i < array.length; i++) {
+		const item = {
+			...array[i],
+			_index: i,
+			_level: level,
+			_size: array.length,
+		} as unknown as T;
+
+		if (Array.isArray(array[i]![nestedKey])) {
+			delete item[nestedKey];
+			yield item;
+			yield* flatten(array[i]![nestedKey], nestedKey, level + 1);
+		} else {
+			yield item;
+		}
+	}
+}
```

Comment:
Well I'm still not sure about using a [generator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator) here but this is very good for traversing a tree because we can pause and return later and this is great for when we need to phase the iteration when we have a lot of data, now we iterate through the entire tree to put together the list but we can take better advantage of this when we have more use cases.

- Phase the iteration to avoid blocking the main thread, this can work using a `requestIdleCallback` or using the [browser's scheduler](https://developer.mozilla.org/en-US/docs/Web/API/Scheduler) to yield CPU time to other tasks so we can meet an appropriate deadline following the [RAIL Model](https://web.dev/rail/)
- We can ignore non-visible items and recompute when they are visible (We can make a stress test to see the advantage of this)

Reply:
Yeah, I have the same concern. I wasn't aware of how generators worked so I had to read up on that. It's not that complicated after reading up on it though, so I think it's probably okay. We should be aware of it though, and consider changing it out for something simpler if we don't end up needing it's pause capability.